packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -478,6 +478,14 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     final Set<LogicalKeyboardKey> keysPressed = LogicalKeyboardKey.collapseSynonyms(RawKeyboard.instance.keysPressed);
     final LogicalKeyboardKey key = keyEvent.logicalKey;
 
+    final bool isWeb = keyEvent.data is RawKeyEventDataWeb;
+    if (isWeb) {
+      // On web platform, We should ignore movement keys and delete key because it's processed already.
+      if(_movementKeys.contains(key) || key == LogicalKeyboardKey.delete) {
```

Comment:
The reason we decided to let the browser handle all key events for us is because it's difficult to hardcode the handling of all keys.

For example, movement keys and delete are not the only ones we should ignore here. What about backspace, home, end, page-up, page-down, enter, etc?

Could you explain your use-case a bit more? Like, why do you need to listen to keys when the user is editing a text field?

Reply:
Hello thank you for your comment, :)

First, regarding the keys we should ignore on control,
**Duplicated operation on web is related to movement keys and delete key only** because it's handled in this _handleKeyEvent() function only.
Other keys(backspace, home, end, page-up...etc) are not handled here thus the duplicated operation doesn't happen. At least as far as I know. if there is anything I missd, please let me know.

Use-case for me is,
1. I hope to use ctrl+c, ctrl+v for copy and paste.
2. I need short-cut key. For example I popup dialog when the user press short-cut key on edit control.
3. I want to make custom rich editor control in future. I need receive key event on web for it. It's possible on Android or PC platform.

Reply:
Sorry for the delay. Yeah, those use-cases make sense.